pkg/types/validation/installconfig.go

```diff
@@ -566,8 +566,13 @@ func validateNamedRepository(r string) error {
 		// in an error. Instead we will check whether the input is
 		// a valid hostname as a workaround.
 		if err == dockerref.ErrNameNotCanonical {
-			_, err := url.ParseRequestURI(r)
+			// If the hostname string contains a port, lets attempt
+			// to split them
+			host, _, err := net.SplitHostPort(r)
 			if err != nil {
+				host = r
```

Comment:
Can you add a short comment describing that this case is when the repository does not contain a port? It requires a bit of thinking when reading this to understand why we can simply ignore the error.

Reply:
Done. Thanks.